src/avt/VisWindow/Colleagues/avtTimeSliderColleague.C

```diff
@@ -712,8 +713,46 @@ avtTimeSliderColleague::SetText(const char *formatString, const char *timeFormat
         strcpy(timeFormatString, timeFormat);
     }
 
+    //
+    // The two variables here are poorly named. One is the textFormatString
+    // which holds the text part of the string like "Time=" or "Cycle=", etc.
+    // The other is timeFormatString which holds a printf-style format
+    // string indicating how to format the numerical value associated with
+    // time.
+    //
+
+    //
+    // We have two possible places the printf-format string could be specified.
+    // One is part of the textFormatString itself (the new way) as in
+    // "Time=$time%6.4f" and the other (the old way) is in the separate
+    // timeFormatString. Which to use?
+    //
+
+    // 
+    // We first check for possible printf-style format string within the
+    // textFormatString itself. If we find something string there, we just
+    // allow it to proceed. If we do not find a format string there but do
+    // find a non-empty, non-default value in timeFormatString, then we
+    // temporarily create a textFormatString with the timeFormatString
+    // appended to it to pass on to create the annotation string.
+    //
+
     delete [] textString;
-    textString = CreateAnnotationString(textFormatString);
+    char const *rawStrWithFormatPattern = R"(\$[a-z]*%[.\d]+[GgFfEeAa])";
+    int formatInTextString = StringHelpers::FindRE(textFormatString,
+                                 rawStrWithFormatPattern);
+    if (formatInTextString == StringHelpers::FindNone &&
+        tf_len > 0 &&
+        std::string(timeFormatString) != "%g")
+    {
+        std::string combined = std::string(textFormatString) +
+                               std::string(timeFormatString);
+        textString = CreateAnnotationString(combined.c_str());
+    }
+    else
+    {
```

Comment:
c++'s fmt lib deals with complex formatting with more embedded markup.

Here is a general example:

https://github.com/LLNL/conduit/blob/0d8db6d00441d97a095f817bd416c3e04eec153c/src/tests/conduit/t_conduit_utils.cpp#L720

Instead of implement a lot of logic ourselves, we might want to use format to help with complex cases.

Reply:
Please have a look at this [documentation](https://github.com/visit-dav/visit/blob/0861518abf80bf16a967495fd12526623d437adb/src/avt/VisWindow/Colleagues/avtAnnotationWithTextColleague.C#L229) and this [comment](https://github.com/visit-dav/visit/blob/0861518abf80bf16a967495fd12526623d437adb/src/avt/VisWindow/Colleagues/avtAnnotationWithTextColleague.C#L229) in the source code, in particular paying attention to how *default* format is handled. In addition lemme know how printf-format can be handled in cases where we specify multiple, same or different `$macros` in the same annotation string, as well as the ability to handle all the different annotation macros that are now supported, as well as the ability to easily add new ones in the future (which was extremely useful when @ndimensionalspace wanted to take annotation strings from files), and lemme know if you still think it is "unnecesarily complicated". BTW, you approved all that work in #17811 two years ago.

Reply:
If the process of creating the string that gets passed to CreateAnnotationString (and that method itself) works as intended, why do we need to also allow user to specify the printf-format in the text string like this `Time=$time%6.4`?

You also asked about doing away with the field that only specifies the printf-format altogether and forcing it to be part of the text string, which is also what I was attempting to question in my previous comment.

Reply:
> Did you ever add tests for these more complex use-cases? If not I will create ticket.

I did not but wanted do and I honestly think now is best time. So, I will work to add some of that here in this PR.

Reply:
Ok, I've added tests. Now requesting re-review

Reply:
BTW...the tests ran on macOS. Could be different from poodle. If so, will re-base after nightlies run.